pandas/core/generic.py

```diff
@@ -2980,18 +2981,45 @@ def astype(self, dtype, copy=True, raise_on_error=True, **kwargs):
 
         Parameters
         ----------
-        dtype : numpy.dtype or Python type
+        dtype : numpy.dtype, Python type, Pandas extension type, or dict
```

Comment:
Can you make this something like  "data type, or dict of column -> type". 
And then you can sum the different options (numpy.dtype, Python type, pandas extension type) in the explanation below.